File: ruby/raindrops/raindrops.rb
class Raindrops
  CONVERT_RULE = {
      'Pling' => 3,
      'Plang' => 5,
      'Plong' => 7
  }
  def self.convert(num)
    result = ''

    CONVERT_RULE.each_pair do |key, value|
      result = result + key if num % value == 0
    end

    result == '' ? num.to_s : result
  end
end

module BookKeeping
  VERSION = 3
end

File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
)

const testVersion = 2

var rule = []struct {
	number int
	match  string
}{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

func Convert(num int) string {
	var result string

	for _, v := range rule {
		if num%v.number == 0 {
			result = result + v.match
		}
	}

	if result == "" {
		return fmt.Sprintf("%d", num)
	} else {
		return result
	}
}
